### PR TITLE
Add convenience script to run most ci checks locally

### PR DESCRIPTION
**Pull request by heeckhau**
This script can be used to run checks before pushing changes to the repository.
It is a good approximation of what CI will do.

### CODE REVIEW

File: pre-commit-check.sh [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+cargo clippy --all-features --all-targets -- -D warnings
+
+# Build all targets
+cargo build --all-targets
+
+# Run tests
+cargo test
+
+# Run wasm tests
+./crates/wasm-test-runner/run.sh
```
Comment (yuroitaki) [inline pre-commit-check.sh:22]: not running those ignored expensive integration tests?

Comment (heeckhau) [reply pre-commit-check.sh:22]: I was a bit undecided here. I wanted a script to "quickly check" my local branch before pushing to GitHub and avoiding follow-up commits.

The main reason for extra follow-up commits is mostly **formatting** and **clippy**. So maybe we should limit this script to those.

Comment (yuroitaki) [reply pre-commit-check.sh:22]: agreed, we can let testing be done in GH and only use this for clippy and formatting.

**Review by themighty1 — APPROVED** [review]
Thanks, that's a good start. The name of the script is appropriate, too.

**Pull request merged**